test/unit/communication/test_communicator_grid.cpp

```diff
@@ -29,7 +29,7 @@ void test_grid_communication(CommunicatorGrid& grid) {
   if (MPI_COMM_NULL == grid.rowCommunicator() || MPI_COMM_NULL == grid.colCommunicator())
     return;
 
-  const int buffer_send = 1;
+  int buffer_send = 1;
```

Comment:
🤔

```c
int MPI_Allreduce(const void* sendbuf, void* recvbuf, int count, MPI_Datatype datatype, MPI_Op op, MPI_Comm comm) 
```

(source: https://www.mpi-forum.org/docs/mpi-3.1/mpi31-report/node117.htm#Node117)

Reply:
@albestro what you're posting is correct. What clang is saying is dumb. We're basically silencing a bogus warning from clang.

Reply:
Does it happens with newer clang versions as well?

Reply:
I was apparently typing at the same time as @rasolca so my response did not take this into account

> Does it happens with newer clang versions as well?

I don't know. But I can try if I attempt to add some new CI configurations.

Reply:
I'll have a go at https://github.com/eth-cscs/DLA-Future/issues/847 once this and #845 are merged.